app/controllers/api/v1/listen_signals_controller.rb

```diff
@@ -5,36 +5,32 @@ class Api::V1::ListenSignalsController < Api::V1::BaseController
   before_action :ensure_user_can_get_listen_signal, only: [:show]
 
   def update
-    @signal = current_user.brand.listen_signals.where(name: signal_params[:name]).first
+    @listen_signal = ListenSignal.find(params[:id])
+    signal_params = signal_params(params)
+    @listen_signal.update_attributes!(signal_params)
 
-    update_signal(@signal, signal_params)
-    update_responses(@signal, signal_params['firstResponse'], signal_params['repeatResponse'])
+    default_response = update_response(@listen_signal.default_response, params[:default_response])
+    repeat_response = update_response(@listen_signal.repeat_response, params[:repeat_response])
 
-    flash[:success] = "Alert:  #{@signal.name} signal updated!"
-    render json: {
-      signal: @signal.to_json,
-      responses: @signal.responses.to_json
-    }
+    if @listen_signal 
+      render json: @listen_signal, each_serializer: ListenSignalSerializer
+    end
   end
 
   def create
-    @brand = current_user.brand
-    @identity = @brand.identities.first
-
-    name = signal_params['name']
-    signal_type = signal_params['signalType']
-    first_response = signal_params['firstResponse']
-    repeat_response = signal_params['repeatResponse']
-    exp_date = signal_params['expirationDate']
-
-    @signal = ListenSignal.create_signal(@brand, @identity, name, signal_type, exp_date)
-    create_template_response(@signal, first_response, repeat_response, exp_date)
-
-    if @signal
-      render json: {
-        signal: @signal.to_json,
-        responses: @signal.responses.to_json
-      }
+    brand = current_user.brand
+    identity = brand.identities.first
+    signal_params = signal_params(params)
+    signal_params[:brand] = brand
+    signal_params[:identity] = identity
+    @listen_signal = ListenSignal.create!(signal_params)
+
+    default_response_msg = params[:default_response]
+    repeat_response_msg = params[:repeat_response]
+    create_grouped_response(default_response_msg, repeat_response_msg)
+
+    if @listen_signal
+      render json: @listen_signal, each_serializer: ListenSignalSerializer
     else
       flash[:error] = 'Alert:  Please fill out missing fields'
```

Comment:
Another note about raising and rescuing errors: I'd like us to use ApiErrors as the standard way of sending json error messages to the client. See [here](https://github.com/project-signal/project-signal/blob/master/app/controllers/api/v1/base_controller.rb#L7). Notice how I'm using `rescue_from`. What I'd like to do as well is add another rescue from just before that where we rescue `StandardError`, log whatever the error is to Rollbar and then raise `ApiErrors::StandardError` with status 500 that way we are responding with standard format error messages from our API.

As far as catching errors when creating models, I was thinking we could have a generic `rescue_from` that rescues `ActiveRecord::ActiveRecordError` which is a generic ActiveRecord error. For now I'd like to Rollbar it and raise a `ApiErrors::StandardError` with status code 400. It's very likely that it will get very noisy in Rollbar. That's fine for now. Later on we can start excluding certain error types from being sent to Rollbar.